geode-old-client-support/src/main/java/com/gemstone/gemfire/OldClientSupportProvider.java

```diff
@@ -121,11 +123,21 @@ public Throwable getThrowable(Throwable theThrowable, KnownVersion clientVersion
     if (theThrowable == null) {
       return theThrowable;
     }
+
+    String className = theThrowable.getClass().getName();
+
+    // GEODE-9452, backward compatibility for authentication expiration
+    if (clientVersion.isOlderThan(KnownVersion.GEODE_1_14_0)) {
+      if (className.equals(AuthenticationExpiredException.class.getName())) {
+        return new AuthenticationRequiredException("User authorization attributes not found.");
```

Comment:
nice!

Reply:
To preserver consistency with the `AuthenticationRequiredException` perhaps they should both use the constant string here.